### PR TITLE
Fetch participants per competition completed....

### PR DESCRIPTION
**Pull request by mthembbu**

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 1]
```diff
@@ -32,15 +34,15 @@ public IEnumerable<Competition> GetCompetitions()
             return competitionsList;
         }
         /** The method that return an array of competition objects whether status is true or false*/
-        // GET: api/Competition
+        // GET: api/Competition/{all}
         [HttpGet("all")]
         public IEnumerable<Competition> GetAllCompetitions()
        {
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: indentation

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 2]
```diff
@@ -58,14 +60,20 @@ public async Task<Competition> CompetitionGetByName(string Name)
         [HttpPost]
         public async Task<IActionResult> AddCompetition([FromBody]Competition comp)
         {
-            if (!ModelState.IsValid)
+            if (ModelState.IsValid)
             {
-                return BadRequest(ModelState);
+                Competition dbComp = _context.Competitions.FirstOrDefault(C => C.Name == comp.Name);
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: use a lower case letter for these Linq queries

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 3]
```diff
@@ -58,14 +60,20 @@ public async Task<Competition> CompetitionGetByName(string Name)
...
+                if (dbComp != null)
+                    return new BadRequestObjectResult(comp.Name+"already exists");
+
+                _notificationMessages.CompetitionNotification(_context, comp);
+                await _context.AddAsync(comp);
+                await _context.SaveChangesAsync();
+                return Ok("Competition Added!!!!!!!");
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: How do you know for sure that the 2 previous awaits were completed successfully? Perhaps check the success of them first before confirming to the user that it was successful

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 4]
```diff
@@ -113,5 +121,23 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
         public void Delete(int id)
         {
         }
+        //GET: api/competition/participants
+        [HttpGet("participants")]
+        public Dictionary<int, int> getUsersPerCompetition()
+        {
+            Dictionary<int, int> mapCompToNumUser = new Dictionary<int, int>();
+            //var CompListQuery = from score in _context.Scores
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: clean this up

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 5]
```diff
@@ -113,5 +121,23 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
...
+            var competitionsList = this.GetCompetitions();
+
+            foreach (var comp in competitionsList)
+            {
+                int count = (from score in _context.Scores
+                             where score.Competition.Id == comp.Id
+                             select score.User.Id).Distinct().ToList().Count();
+               mapCompToNumUser.Add(comp.Id,count);
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: indentation

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 6]
```diff
@@ -58,14 +60,20 @@ public async Task<Competition> CompetitionGetByName(string Name)
         [HttpPost]
         public async Task<IActionResult> AddCompetition([FromBody]Competition comp)
         {
-            if (!ModelState.IsValid)
+            if (ModelState.IsValid)
             {
-                return BadRequest(ModelState);
+                Competition dbComp = _context.Competitions.FirstOrDefault(C => C.Name == comp.Name);
+                if (dbComp != null)
+                    return new BadRequestObjectResult(comp.Name+"already exists");
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: leave a space between variable and text

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs (hunk 1 above)
Comment (mthembbu) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: Done

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs (hunk 2 above)
Comment (mthembbu) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: Done

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs (hunk 3 above)
Comment (mthembbu) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: Hi Rashad, most backend devs didn't mind for not checking the outcome, and I do understand what you do mean. I am new on the backend, and my LINQ is not as good yet. I will google, and learn how to check the bool for async functions.

Comment (RashadAkoodieRabbit) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: You can just assign a local variable to the result of the async task and check the result and do the handling in this class

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 7]
```diff
@@ -62,9 +62,9 @@ public async Task<IActionResult> AddCompetition([FromBody]Competition comp)
         {
             if (ModelState.IsValid)
             {
-                Competition dbComp = _context.Competitions.FirstOrDefault(C => C.Name == comp.Name);
-                if (dbComp != null)
-                    return new BadRequestObjectResult(comp.Name+"already exists");
+                Competition dbcomp = _context.Competitions.FirstOrDefault(C => C.Name == comp.Name);
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: i meant a small letter for your c => c... your variable was named correctly

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 8]
```diff
@@ -62,9 +62,9 @@ public async Task<IActionResult> AddCompetition([FromBody]Competition comp)
...
+                if (dbcomp != null)
+                    return new BadRequestObjectResult(comp.Name +" already exists");
```
Comment (RashadAkoodieRabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: space after "+"

**Pull request merged**